fish-rust/src/wgetopt.rs

```diff
@@ -451,18 +451,20 @@ impl<'opts, 'args, 'argarray> wgetopter_t<'opts, 'args, 'argarray> {
         // Test all long options for either exact match or abbreviated matches.
         for (option_index, p) in self.longopts.iter().enumerate() {
             if p.name.starts_with(&self.nextchar[..nameend]) {
-                // Exact match found.
-                pfound = Some(*p);
-                *indfound = option_index;
-                *exact = true;
-                break;
-            } else if pfound.is_none() {
-                // First nonexact match found.
-                pfound = Some(*p);
-                *indfound = option_index;
-            } else {
-                // Second or later nonexact match found.
-                *ambig = true;
+                if nameend == p.name.len() {
+                    // Exact match found.
+                    pfound = Some(*p);
+                    *indfound = option_index;
+                    *exact = true;
+                    break;
+                } else if pfound.is_none() {
+                    // First nonexact match found.
+                    pfound = Some(*p);
+                    *indfound = option_index;
+                } else {
+                    // Second or later nonexact match found.
+                    *ambig = true;
+                }
```

Comment:
Would be nice to have this in a separate commit with some explanation, it's not immediately obvious what's happening here.

Reply:
I am also not sure what is happening here, I found the issue by comparing this with actual master CPP code and found the logic issue. I can put some explanation on what I encountered and explain

Reply:
Have updated the comment let if know what you think